design-proposals/klip-4-produce-data.md

```diff
@@ -0,0 +1,302 @@
+section that includes the following rows:
+> ```rst
+> +=========================+======================================================================================================+
+> | KAFKA_TOPIC (required)  | The name of the Kafka topic that backs this source. The topic must either already exist in Kafka, or |
+> |                         | PARTITIONS and REPLICAS must be specified to create the topic. Command will fail if the topic exists |
+> |                         | with different partition/replica counts.                                                             |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | PARTITIONS              | The number of partitions in the backing topic. This property must be set if creating a SOURCE without|
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> | REPLICAS                | The number of replicas in the backing topic. This property must be set if creating a SOURCE without  |
+> |                         | an existing topic.                                                                                   |
+> +-------------------------+------------------------------------------------------------------------------------------------------+
+> ```
+* A new section for `INSERT INTO ... VALUES` will be added to the `sytnax.rst`:
+> ```rst
+> **Synopsis**
+> 
+> .. code:: sql
+>
+>   INSERT INTO <stream_name|table_name> [(column_name [, ...] )]
+>     VALUES (value [, ...])
+> 
+> **Description**
+> 
+> Produce a row into an existing stream or table and its underlying topic based on
+> explicitly sepcified values. The first ``column_name`` of every schema is ``ROWKEY``, which
+> defines the corresponding kafka key - it is expected to always match a corresponding column in 
+> the value if the create statement specifies a key.
+>
+> Column values can be given in several ways:
+> * Any column not explicitly given a value is set to ``null`` (KSQL does not support DEFAULT values)
+> * If no columns are specified, a value for every column is expected in the same order as the
+> schema
+>
+> The values will serialize using the ``value_format`` specified in the original `CREATE` statment.
+> ```
+* A new section for `DELETE` will be added to the `syntax.rst`:
+> ```rst
+> **Synopsis**
+> 
+> .. code:: sql
+> 
+>     DELETE FROM <table_name> WHERE ROWKEY = <value>; 
+>     
+> **Description**
+> 
+> Delete a row from an existing table. This will issue a tombstone, producing a value to the 
+> underlying kafka topic with the specified key and a null value. Deleting a value that 
+> does not exist will not affect the table contents.
```

Comment:
What happens if someone tries to issue this statement on a stream? I assume they'll get an error? Perhaps worth slipping a short sentence about that into the docs here.